src/types.h

```diff
@@ -159,6 +159,7 @@ class SizedType {
   bool ctx_ = false;                                   // Is bpf program context
   std::unordered_set<std::string> btf_type_tags_ = {}; // Only populated for
                                                        // Type::pointer
+  size_t num_elements_ = 0; // Only populated for array types
```

Comment:
See `CreateArray()`. It looks like this information is already present, except perhaps not super clearly. Basically `num_elements_` exists as `size_bits_ / 8 / element_type_->GetSize()`. I think we should only have one way to encode array size.

Reply:
Scratch that - looks like you did exactly this :)